### ISSUE TITLE
No output when using .eslintrc with versions 1.0.0-rc-1 and above

### ISSUE THREAD

**Issue by reviling** [eslint/eslint#3248]
TL;DR: Using a .eslintrc file with any settings works in version 24.1 will output to the console all the various errors. Using versions 1.0.0-rc-1 and above will output nothing to the console.

This is my .eslintrc file:
```javascript
{
  "globals": {
    "var1": true,
    "var2": false
  }
}
```
If I install 24.1, things work.
```
$ sudo npm install -g eslint@0.24.1
/usr/local/bin/eslint -> /usr/local/lib/node_modules/eslint/bin/eslint.js
eslint@0.24.1 /usr/local/lib/node_modules/eslint
├── escape-string-regexp@1.0.3
├── object-assign@2.1.1
├── path-is-absolute@1.0.0
├── xml-escape@1.0.0
├── user-home@1.1.1
├── strip-json-comments@1.0.4
├── estraverse-fb@1.3.1
├── globals@8.3.0
├── estraverse@4.1.0
├── text-table@0.2.0
├── espree@2.2.3
├── debug@2.2.0 (ms@0.7.1)
├── chalk@1.1.0 (supports-color@2.0.0, ansi-styles@2.1.0, has-ansi@2.0.0, strip-ansi@3.0.0)
├── doctrine@0.6.4 (isarray@0.0.1, esutils@1.1.6)
├── mkdirp@0.5.1 (minimist@0.0.8)
├── optionator@0.5.0 (type-check@0.3.1, deep-is@0.1.3, levn@0.2.5, wordwrap@0.0.3, prelude-ls@1.1.2, fast-levenshtein@1.0.6)
├── is-my-json-valid@2.12.1 (jsonpointer@1.1.0, generate-function@2.0.0, xtend@4.0.0, generate-object-property@1.2.0)
├── concat-stream@1.5.0 (inherits@2.0.1, typedarray@0.0.6, readable-stream@2.0.2)
├── minimatch@2.0.10 (brace-expansion@1.1.0)
├── js-yaml@3.3.1 (esprima@2.2.0, argparse@1.0.2)
├── inquirer@0.8.5 (figures@1.3.5, ansi-regex@1.1.1, cli-width@1.0.1, through@2.3.8, readline2@0.1.1, lodash@3.10.0, rx@2.5.3)
└── escope@3.2.0 (esrecurse@3.1.1, estraverse@3.1.0, es6-weak-map@0.1.4, es6-map@0.1.1)

$eslint ./app/webroot/js/test.js

./app/webroot/js/test.js
    1:16  error  Missing "use strict" statement                           strict

✖ 142 problems (142 errors, 0 warnings)

$
```
If I install 1.0.0-rc-1 and above, things don't work. :(
```
$ sudo npm install -g eslint@1.0.0-rc-1
/usr/local/bin/eslint -> /usr/local/lib/node_modules/eslint/bin/eslint.js
eslint@1.0.0-rc-1 /usr/local/lib/node_modules/eslint
├── escape-string-regexp@1.0.3
├── path-is-absolute@1.0.0
├── object-assign@2.1.1
├── path-is-inside@1.0.1
├── xml-escape@1.0.0
├── user-home@1.1.1
├── strip-json-comments@1.0.4
├── estraverse-fb@1.3.1
├── globals@8.3.0
├── estraverse@4.1.0
├── text-table@0.2.0
├── espree@2.2.3
├── debug@2.2.0 (ms@0.7.1)
├── doctrine@0.6.4 (isarray@0.0.1, esutils@1.1.6)
├── mkdirp@0.5.1 (minimist@0.0.8)
├── optionator@0.5.0 (type-check@0.3.1, deep-is@0.1.3, levn@0.2.5, wordwrap@0.0.3, prelude-ls@1.1.2, fast-levenshtein@1.0.6)
├── chalk@1.1.0 (supports-color@2.0.0, ansi-styles@2.1.0, strip-ansi@3.0.0, has-ansi@2.0.0)
├── is-my-json-valid@2.12.1 (jsonpointer@1.1.0, generate-function@2.0.0, xtend@4.0.0, generate-object-property@1.2.0)
├── concat-stream@1.5.0 (inherits@2.0.1, typedarray@0.0.6, readable-stream@2.0.2)
├── minimatch@2.0.10 (brace-expansion@1.1.0)
├── inquirer@0.8.5 (figures@1.3.5, ansi-regex@1.1.1, cli-width@1.0.1, through@2.3.8, lodash@3.10.0, readline2@0.1.1, rx@2.5.3)
├── js-yaml@3.3.1 (esprima@2.2.0, argparse@1.0.2)
└── escope@3.2.0 (esrecurse@3.1.1, estraverse@3.1.0, es6-weak-map@0.1.4, es6-map@0.1.1)

$ eslint ./app/webroot/js/mhe.helper.js
$
```
If I make my .eslintrc file look like the following, things will "work". Just not with any settings of course.
```javascript```
{
}
```
Essentially, if any configuration is set, nothing will be outputted to the console. I have tried this with every version above 1.0.0-rc-1 and have the same issue.

I am on...
OSX v10.10.4
Node v0.12.7

**Comment by btmills** [comment]
We don't enable any rules by default anymore in 1.0. If you want some of the same behavior, try adding `"extends": "eslint:recommended"` to your eslintrc. That will enable our recommended set of rules to catch potential errors, but you'll want to set stylistic rules yourself.

**Comment by reviling** [comment]
Aha! That did indeed do the trick. I feel as though the documentation needs to be updated to reflect this fact.

**Comment by nzakas** [comment]
It was mentioned in the blog: http://eslint.org/blog/2015/07/eslint-1.0.0-released/#reset-is-now-the-default

And also in the migration guide: http://eslint.org/docs/user-guide/migrating-to-1.0.0#all-rules-off-by-default

**Comment by reviling** [comment]
That is not enough. The main documentation eludes to how it was configured before, which will cause confusion like it did with me. I didn't even know the blog or migration guide existed until you mentioned it here. You should not expect new users to look beyond the user-guide as that should be expected to be kept up to date, which it is not.

Bad examples from: http://eslint.org/docs/user-guide/configuring
* >ESLint is designed to be completely configurable, meaning you can turn off every rule and run only with basic syntax validation, or mix and match the bundled rules and your custom rules to make ESLint perfect for your project

* >By default, ESLint supports only ECMAScript 5 syntax.

* >By default, ESLint uses Espree as its parser.

* >There's no need to specify every single rule - you will automatically get the default setting for every rule. You only need to override the rules that you want to change. Note: All rules that are enabled by default are set to 2, so they will cause a non-zero exit code when encountered. You can lower these rule to a warning by setting them to 1, which has the effect of outputting the message onto the console but doesn't affect the exit code.

**Comment by IanVS** [comment]
Ah, it seems the `1.0.0` version of the user-guide didn't get moved over. http://eslint.org/docs/1.0.0/user-guide/configuring

**Comment by IanVS** [comment]
The first three bullets are still true, but I think you're right the wording in the last should be updated.

**Comment by reviling** [comment]
>meaning you can turn off every rule and run only with basic syntax validation

This implies there are initial rules you can turn off, which there are not.

The second two point implies that reset doesn't reset everything, which is contradictory to what I would have expected. Where can I find details on the default settings?

Thanks.

**Comment by IanVS** [comment]
There still needs to be a parser, `reset` never affected it.  Same with es5, that is still the default unless you enable es6 features.  As you can see from the [0.24.1 docs](http://eslint.org/docs/0.24.1/user-guide/command-line-interface#reset), reset only applied to rules.  Syntax checking isn't part of the rules, it looks for invalid JavaScript which would crash if you tried to execute it.  That will still happen in `1.0.0`, so I think that comment is okay.

**Comment by reviling** [comment]
Thanks. That clears everything up for me. :)
